extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/OidcClientRequestFilter.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.oidc.common;
+
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+
+/**
+ * Request filter which can be used to customize OIDC client requests
+ */
+public interface OidcClientRequestFilter {
+    /**
+     * Filter OIDC client requests
+     *
+     * @param request HTTP request
+     * @param body request body, will be null for HTTP GET methods, may be null for other HTTP methods
+     */
+    void filter(HttpRequest<Buffer> request, Buffer body);
```

Comment:
Isn't there a flag or something that could be used to prevent further processing? Like `javax.servlet.FilterChain`?

Reply:
@gastaldi This is more in line with REST client filters approach, they are run one by one, as opposed to the first filter calling some context to proceed, does it clarify it ?